website/blog/unusual-builds-w-bytes.md

```diff
@@ -0,0 +1,217 @@
+
+Once we had [upgraded to a Bazel version](./blog/bisect-bazel) with this change, we were able to see that the unwanted downloads were coming from the `output` downloads.
+
+# The Bug
+
+We discovered another code path in Bazel that can trigger unexpected downloads: when a blob's TTL expires.
+
+> So... What is a blob's TTL? 🤔
+
+As an artifact-oriented build system, Bazel relies heavily on artifacts existing in its output path to determine if a build is up-to-date.
+If an artifact exists and matches the expected result, then Bazel will skip the action that produces the artifact.
+However, in a remote build with minimal download, Bazel does not have the artifact locally to validate against and must establish some trust that the artifact was created and stored in the Remote Cache.
+
+Introduced in [PR#17639](https://github.com/bazelbuild/bazel/pull/17639), this "trust" is added to Bazel by storing the expected Time To Live (TTL) of the artifact in the Remote Cache.
+When Bazel executes an action remotely, instead of downloading the outputs referenced inside the ActionResult, it instead just stores the references inside its in-process analysis cache (aka. SkyFrame) with an expected TTL that is determined by the flag `--experimental_remote_cache_ttl` (default: 3 hours).
+When the TTL expires, instead of checking if the output artifacts are up-to-date remotely, Bazel downloads the entire artifact to disk.
+We were able to validate this by setting the flag `--experimental_remote_cache_ttl=0s` to eagerly trigger the downloads.
+
+```bash
+# Initial build
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=1d
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+1.8M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+
+# Incremental build
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=1d
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+1.8M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+
+# Incremental build with TTL expired
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=0
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+640M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+```
+
+There was a draft fix by [David Sanderson](https://github.com/dws) in [PR#23066](https://github.com/bazelbuild/bazel/pull/23066), however, it was incomplete and never reviewed.
+Instead of introducing another flag and asking users to configure Bazel correctly, we decided to submit a different fix in [PR#25398](https://github.com/bazelbuild/bazel/pull/25398) with a sensible default: don't download output artifacts once the TTL expires. Instead, just discard the blob metadata with the expired TTL and re-execute the action if needed.
```

Comment:
I wouldn't word this as being about a default. I think that the bug was introduced simply due to the confusing naming of the function that checks the metadata (not something about the artifact). I don't think there is ever a case where you would want to download artifacts that you believe may no longer be alive - they might very well not even exist anymore.

Reply:
I rewrote this section to make it clear that (a) it's a simple bug fix and (b) the draft community PR could potentially be a work around.